Solutions/Corvus.SpecFlow.Extensions.Specs/Containers/PerFeatureContainer.feature

```diff
@@ -0,0 +1,37 @@
+﻿# These tests are a bit meta, because they test that the library enables us to write SpecFlow specs in the way we want,
+# and they do so through the medium of SpecFlow specs. In some cases this is straightforward, but it makes certain
+# aspects of the test look a little odd.
+
+# This tag is essentially the thing we're testing. Tests using per-feature containers will put this tag at the top of
+# their files.
+@perFeatureContainer
+
+# This tag required for these tests to work. It enables before/after bindings that perform work needed to test
+# the behaviour of the @perFeatureContainer feature.
+@runPerFeatureContainerTests
+
+Feature: PerFeatureContainer
+	In cases where my tests need a DI container that persists across all the scenarios in a feature
+	As a developer
+	I want to be able to define container services in a BeforeFeature binding and then use those services later in the test
+
+Background:
+	Given I have specified the perFeatureContainer tag
+
+Scenario: Services added during PopulateServiceCollection BeforeFeature phase are in Service Provider available to tests
+    Given I use feature ContainerBindings.GetServiceProvider during a Given step
+	When I use feature ContainerBindings.GetServiceProvider during a When step
+    Then if I also use feature ContainerBindings.GetServiceProvider during a Then step
```

Comment:
Clearly I'm approaching this in the wrong order, as I've just added this exact same comment to the `PerFeatureContainer` file...

Since lines 22-24 are essentially the "Act" parts of the spec, it would make a lot more sense to me if they were prefixed "When", "And", "And" rather than "Given", "When", "Then".

Reply:
As with the other file, how would that actuall work? They'd all run during the `When` phase, at which point they'd be failing to test the scenario I want to test.